src/eduid_common/config/parsers/etcd.py

```diff
@@ -5,29 +5,25 @@
 import json
 import logging
 import os
+from typing import Any, Mapping, Optional
 
 import etcd
 
+from eduid_common.config.parsers.base import BaseConfigParser
 from eduid_common.config.parsers.decorators import decrypt, interpolate
 from eduid_common.config.parsers.exceptions import ParserException
 
 __author__ = 'lundberg'
 
 
-class EtcdConfigParser(object):
-    def __init__(self, namespace, host=None, port=None):
+class EtcdConfigParser(BaseConfigParser):
+    def __init__(self, namespace: str, host: Optional[str] = None, port: Optional[int] = None, silent: bool = False):
         """
         :param namespace: etcd namespace to read or write, ex. /eduid/webapp/common/
         :param host: Optional etcd host
         :param port: Optional etcd port
-
-        :type host: str | unicode
-        :type namespace: str | unicode
-        :type port: int
-
-        :return: EtcdConfigParser object
-        :rtype: EtcdConfigParser
-        """
+        :param silent: set to `True` if you want silent failure for missing keys.
```

Comment:
I investigated what the code does and updated the comment to

        :param silent: set to `True` if you want to silently ignore etcd errors (such as EtcdConnectionFailed).

but honestly, it sounds a bit crazy to ignore such errors, so maybe we should rather remove this?

Reply:
It is because you can use more than one way to get configuration, so load errors can be ignored. The application should not start if the needed configuration hasn't been loaded.